apps/microsoft-teams/frontend/src/components/config/ContentTypeSelectionModal/ContentTypeSelectionModal.tsx

```diff
@@ -31,10 +33,32 @@ const ContentTypeSelectionModal = (props: Props) => {
   } = props;
 
   const [selectedContentTypeId, setSelectedContentTypeId] = useState(savedContentTypeId ?? '');
-
+  const [filteredContentTypes, setFilteredContentTypes] =
+    useState<ContentTypeProps[]>(contentTypes);
   const { title, button, link, emptyContent, emptyHeading, errorMessage } =
     contentTypeSelection.modal;
 
+  const handleSearch = useCallback(async (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.value === '') {
+      // revert to default full list of available contentTypes
+      setFilteredContentTypes(contentTypes);
+      return;
+    }
+
+    const fuseOptions = {
+      isCaseSensitive: false,
+      keys: ['sys.id', 'displayField', 'name', 'description'],
+    };
+
+    const fuse = new Fuse(filteredContentTypes, fuseOptions);
+    const searchPattern = e.target.value;
+    const fuseSearchResultObj = fuse.search(searchPattern);
+    // extract actual contentType objects from fuse search result object;
+    const extractedSearchResults = fuseSearchResultObj.map((result) => result.item);
```

Comment:
Open to suggestions for a better name than `extractedSearchResults`.  Basically the object that gets returned is a fuse object with `{....todo....}`